src/meltano/core/utils/__init__.py

```diff
@@ -170,6 +170,14 @@ def merge(src, dest):
     return dest
 
 
+def are_similar_types(left, right):
+    return (
+        issubclass(left.__class__, right.__class__)
+        or issubclass(right.__class__, left.__class__)
+        or type(left) == type(right)  # noqa: WPS516
+    )
```

Comment:
Types should generally be compared using `is`, but in this case the type comparison isn't necessary since `issubclass(class_x, class_x)` is `True`.

That said, we can use `isinstance` plus `type` here to avoid accessing the dunder attribute `__class__`.

```suggestion
def are_similar_types(left, right):
    return isinstance(left, type(right)) or isinstance(right, type(left))
```

Reply:
TIL`isinstance` caters for inheritance, while checking for equality of `type` does not 😅 Thanks @WillDaSilva !